client/src/main/java/io/apicurio/registry/client/auth/KeycloakAuth.java

```diff
@@ -0,0 +1,23 @@
+package io.apicurio.registry.client.auth;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+
+public class KeycloakAuth implements AuthStrategy {
+
+    private Keycloak keycloak;
+
+    public KeycloakAuth(AuthConfig config) {
+        this.keycloak = KeycloakBuilder.builder()
+                .serverUrl(config.getServerUrl())
+                .username(config.getUsername())
+                .password(config.getPassword())
+                .realm(config.getRealm())
+                .clientId(config.getClientId())
```

Comment:
How about clientSecret? I try with the confidential clients (keycloak) and I got 400
Bad Request 
--------------------------------------------------------------
error_description | "Client secret not provided in request"
-- | --

Reply:
Yeah - client secret will likely be needed.

Reply:
Yes, someone how I missed adding it.       `.clientSecret(config.getClientSecret())`
Do we also need to allow setting up the ` .grantType()`?